scripts/build_images/build.py

```diff
@@ -0,0 +1,205 @@
+import hashlib
+import json
+import os
+import subprocess
+import tempfile
+from contextlib import nullcontext
+
+from .utils import get_env_or_exit, local_registry, print_and_run_command
+
+# The container name for the local Docker registry.
+REGISTRY_NAME = "prairielearn-registry"
+
+# A mapping from images to their base images. We use this to know to rebuild images when their base image has changed.
+BASE_IMAGE_MAPPING = {
+    # Core images.
+    "prairielearn/prairielearn": "prairielearn/plbase",
+    "prairielearn/executor": "prairielearn/prairielearn",
+    # Workspace images.
+    "prairielearn/workspace-vscode-python": "prairielearn/workspace-vscode-base",
+    "prairielearn/workspace-vscode-cpp": "prairielearn/workspace-vscode-base",
+}
+
+BASE_IMAGES = list(set(BASE_IMAGE_MAPPING.values()))
+
+images = get_env_or_exit("IMAGES")
+tag = get_env_or_exit("TAG")
+metadata_dir = os.environ.get("METADATA_DIR")
+should_push = os.environ.get("PUSH_IMAGES", "false").lower() == "true"
+
+if not metadata_dir:
+    print("No manifest directory specified; manifests will not be created.")
+
+
+def get_image_path(image: str) -> str:
```

Comment:
Can we add an argument for a folder to build from in the script? Then we can avoid this  logic, and only special case `prairielearn/prairielearn`.

Reply:
The above might make running locally worse though since you need to specify it?

Reply:
The script is invoked on batches of images, not just a single one, so you'd need parallel lists of names + directories. This felt simpler and more robust to me.

Reply:
I thought about it, and I agree now that this would get in the way of local running.